### ISSUE TITLE
JSON

### ISSUE THREAD

**Issue by moonglum** [frodsan/tynn#18]
Tynn has a JSON plugin. It however only handles outgoing data by providing the `json` method. Should we also add support for incoming data when the content type is set accordingly? I think the best place for that would be the JSON plugin :smile:

**Comment by frodsan** [comment]
Yes, I would like to add back this feature. This was the implementation before release:

https://github.com/frodsan/tynn/blob/32c8debe381ca77427339ffc74f3ffff6c29eb44/lib/tynn/json_parser.rb

I removed because I wasn't sure about using a Rack Middleware and handling JSON errors
at the same time. But I would be nice to discuss it.

**Comment by moonglum** [comment]
Can you elaborate on what the problem was? :smile:

**Comment by frodsan** [comment]
Yes, I think it would be nice to rescue and do something if the parsing fails:

```
begin
  ... = JSON.parse(...)
rescue x
  return [400, {}, []]
end
```

Is that a good way to response if it fails? We can keep it like that for now and then see how users can customize the response.

**Comment by moonglum** [comment]
Ah, ok. Now I understand! Thanks :smile: 

Yes, I think it should return a 400. I would agree that the body of this response should be customiazable. Some people like to leave it empty, some put a "Not parseable" in there. Some prefer a JSON object.

**Comment by britishtea** [comment]
I think [`415 Unsupported Media Type`](http://devdocs.io/http/rfc7231#section-6.5.13) describes best what went wrong.
